internal/pkg/skuba/kubernetes/node_version.go

```diff
@@ -179,29 +178,39 @@ func nodeVersioningInfo(client clientset.Interface, nodeName string) (NodeVersio
 
 	// find out the container image tags, depending on the role of the node
 	if IsControlPlane(nodeObject) {
-		apiServerTag, err := getPodContainerImageTag(client, metav1.NamespaceSystem, fmt.Sprintf("%s-%s", "kube-apiserver", nodeName))
+		// list all the pods
+		allPods, err := client.CoreV1().Pods(metav1.NamespaceSystem).List(metav1.ListOptions{})
+		// check for error
 		if err != nil {
-			return NodeVersionInfo{}, errors.Wrap(err, "could not retrieve apiserver pod")
+			return NodeVersionInfo{}, errors.Wrap(err, "Could not retrieve pods")
 		}
-		controllerManagerTag, err := getPodContainerImageTag(client, metav1.NamespaceSystem, fmt.Sprintf("%s-%s", "kube-controller-manager", nodeName))
+		// check for empty pod list
+		if len(allPods.Items) == 0 {
+			return NodeVersionInfo{}, errors.Wrap(err, "List of pods is empty")
```

Comment:
errors are typically started with lowercase :), applies to all errors on this PR

Reply:
Besides this, `errors.Wrap` will return `nil` if `err` is `nil` (this is the case here), so you need to create a new error here instead of wrapping.

Reply:
> errors are typically started with lowercase :), applies to all errors on this PR

Thats so weird. Fixed.

> Besides this, errors.Wrap will return nil if err is nil (this is the case here), so you need to create a new error here instead of wrapping.

Fixed!

Reply:
it's a golang thing https://www.reddit.com/r/golang/comments/241u95/best_practices_for_errors_in_go/